src/server.rs

```diff
@@ -32,66 +35,54 @@ impl RocketServer {
     }
 }
 
+#[async_trait]
 impl Server for RocketServer {
-    fn run(self: Box<Self>, address: &str, port: u16) -> Result<(), Box<dyn Error>> {
-        let config = Config::build(Environment::Staging)
-            .address(address)
-            .port(port)
-            .finalize()
-            .map_err(Box::new)?;
+    async fn run(self: Box<Self>, address: IpAddr, port: u16) -> Result<(), Box<dyn Error>> {
+        let config = Config { address, port, ..Default::default() };
 
-        rocket::custom(config)
+        _ = rocket::custom(config)
             .mount("/", routes![search, item_by_slug])
             .manage(self.search_engine)
-            .launch();
+            .launch()
+            .await?;
```

Comment:
why not return this?

Reply:
Instead of Ok(())

Reply:
`launch()` returns a Result containing a `Rocket` instance, our Server trait requires that we return a unit.

Reply:
Could do `[...].await.ok().map(|_|())`. I don't think it's actually better, just wanna point it out.